libcentrifugo/application.go

```diff
@@ -498,7 +498,7 @@ func (app *Application) pubPing() error {
 		Goroutines: runtime.NumGoroutine(),
 		NumCPU:     runtime.NumCPU(),
 		Gomaxprocs: runtime.GOMAXPROCS(-1),
-		Metrics:    *app.metrics,
+		Metrics:    *app.metrics.GetSnapshot(),
```

Comment:
Ah I see so just taking the current value and copying it causes non-atomic reads of the counters which races with atomic increments...

I still don't understand why that would actually cause a nil pointer or invalid memory reference as in the original report...